src/dfcx_scrapi/tools/search_util.py

```diff
@@ -497,6 +533,360 @@ def find_list_parameters(self, agent_id):
 
         return params_list
 
+    def get_param_presets_df(self, flow_name_list=None):
+        """Gets a dataframe of parameter presets.
+
+        Args:
+          flow_name_list (optoinal): list of names of flows to search.
+          By default all flows will be searched.
+
+        Returns:
+          Dataframe of parameter presets with columns:
+            flow,
+            page,
+            intent,
+            event,
+            route_group,
+            condition,
+            form_filling,
+            location,
+            param_name,
+            param_value
+        """
+
+        flow_id_list = []
+        if flow_name_list is None:
+            flow_id_list = list(self.flows_map.values())
+        elif isinstance(flow_name_list, list):
+            for flow_name in flow_name_list:
+                flow_id_list.append(self.flows_map[flow_name])
+        elif isinstance(flow_name_list, str):
+            if flow_name_list in self.flows_map:
+                flow_id_list = list(self.flows_map[flow_name_list])
+            else:
+                raise TypeError("Flow not found")
+        else:
+            raise TypeError(
+                "flow_name_list parameter needs to be a list of flow names"
+            )
+
+        param_dfs = []
+        for flow_id in flow_id_list:
+            #Start Page
+            start_page = self.flow_data[flow_id]
+            flow_name = start_page.display_name
+            df = self.process_param_presets_in_page(flow_name, start_page)
+            param_dfs.append(df)
+            #All Other Pages
+            for page in self.page_data[flow_id].values():
+                #page_name = page.display_name
+                df = self.process_param_presets_in_page(flow_name, page)
+                param_dfs.append(df)
+        #Combine Lists to DataFrame
+        if len(param_dfs) > 0:
+            return pd.concat(param_dfs).reset_index(drop=True)
```

Comment:
The nested return followed by `return None` could be refactored much more cleanly.

After you check the `params_dfs` length, if it passes, set the results of `pd.concat...` to a variable and then simply return that variable. this eliminates the need for 2 return statements.

```py
for flow id in flow_id_list:
   ....
   for page in self.page_data:
    .....

if len(param_dfs) > 0:
   final_df = pd.concat(....)

return final_df
```